pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -342,6 +349,23 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, err
 	}
 
+	// Set region label on the ClusterDeployment
+	if region := getClusterRegion(cd); cd.Spec.Platform.BareMetal == nil && cd.Labels[hivev1.HiveClusterRegionLabel] != region {
+		if cd.Labels == nil {
+			cd.Labels = make(map[string]string)
+		}
+		if cd.Labels[hivev1.HiveClusterRegionLabel] != "" {
+			cdLog.Warnf("changing the value of %s from %s to %s", hivev1.HiveClusterRegionLabel,
+				cd.Labels[hivev1.HiveClusterRegionLabel], region)
+		}
+		cd.Labels[hivev1.HiveClusterRegionLabel] = region
+		err := r.Update(context.TODO(), cd)
+		if err != nil {
+			cdLog.WithError(err).Log(controllerutils.LogLevel(err), "failed to set cluster region label")
```

Comment:
Not familiar with this LogLevel(err) syntax, but in Hive codebase this should probably be cdLog.WithError(err).Error("failed to set cluster region label").

Reply:
No, using `controllerutils.LogLevel(err)` is correct here. This logs update errors that are not controller errors--such as conflict errors--at the info level instead of the error level.